pkg/controller/pxc/upgrade.go

```diff
@@ -225,18 +227,23 @@ func (r *ReconcilePerconaXtraDBCluster) applyNWait(cr *api.PerconaXtraDBCluster,
 		return fmt.Errorf("failed to wait pod: %v", err)
 	}
 
-	if err := r.waitUntilOnline(cr, sfs.Name, pod, waitLimit); err != nil {
-		return fmt.Errorf("failed to wait pxc status: %v", err)
-	}
-
 	if err := r.waitPXCSynced(cr, pod.Status.PodIP, waitLimit); err != nil {
 		return fmt.Errorf("failed to wait pxc sync: %v", err)
 	}
 
+	if err := r.waitUntilOnline(cr, sfs.Name, pod, waitLimit); err != nil {
+		return fmt.Errorf("failed to wait pxc status: %v", err)
+	}
+
 	return nil
 }
 
 func (r *ReconcilePerconaXtraDBCluster) waitUntilOnline(cr *api.PerconaXtraDBCluster, sfsName string, pod *corev1.Pod, waitLimit int) error {
+	if cr.Spec.HAProxy.Enabled {
+		time.Sleep(5 * time.Second)
+		return nil
+	}
```

Comment:
It would be great to have a comment describing why we need just to wait for 5 sec having HAProxy